src/v/cluster/controller_backend.cc

```diff
@@ -853,6 +858,16 @@ ss::future<> controller_backend::add_to_shard_table(
       });
 }
 
+ss::future<> controller_backend::insert_to_shard_table(
+  model::ntp ntp, ss::shard_id shard, model::revision_id revision) {
+    vlog(
+      clusterlog.trace, "insert {} to shard table at {}", revision, ntp, shard);
+    return _shard_table.invoke_on_all(
```

Comment:
Do we need to check that insert returns the same value for all cores?

Reply:
Nice catch

Reply:
I removed the check, `update_shard()` returns nothing just like `shard_table::update()` does. It behaves the same way, this removing the necessity for the checks